03/app.js

```diff
@@ -5,17 +5,65 @@ import Category from './Category';
 import Cart from './Cart';
 
 import data from './data.json';
+import Product from './Product';
+import {v4 as uuid} from 'uuid';
 
 class App extends React.Component {
     state = {
         cart: [],
     }
+
+    addToBasket = (data) => {
+        const { cart } = this.state;
+        data.added = true;
+        this.setState({
+            cart: [...cart, data]
+        });
```

Comment:
Lepiej (bezpieczniej) jest utworzyć nowy obiekt  czyli:
```
const newProduct = {...data, added: true};
this.setState({
    cart: [...cart, newProduct];
})
```
Trzeba pamiętać, że przekazywanie obiektu przez parametr to referencja więc zapis `data.added = true` zamienia obiekt poza funkcją, który został przekazany przez parametr.